openapi.yaml

```diff
@@ -0,0 +1,220 @@
+            changed. The latter might be caused by a change to a record
+            container for which the authenticated entity is not authorized to
+            change.
+        404:
+          description: >
+            the requested CVE ID has not been previously published.
+      requestBody:
+        description: The CVE record data to update
+        content:
+          application/json:
+            schema:
+              type: object
+              # add schema for CVE 5.0 format
+  /cve/{cve-id}/{entity-type}/{entity-id}:
+    parameters:
+      - "$ref": "#/components/parameters/cve-id-path"
+      - name: entity-type
+        in: path
+        description: The type of entity (i.e., CNA, ADP)
+        required: true
+        schema:
+          type: string
+          enum:
+            - CNA
+            - ADP
+      - name: entity-id
+        in: path
+        description: The identifier for the entity's container
+        required: true
+        schema:
+          type: string
+          # what is this syntax?
+    get:
+      description: >
+        Retrieves a given container in a CVE record entry.
+      operationId: getCVERecordContainerById
+      responses:
+        200:
+          description: >
+            The requested CVE record container associated with the CVE ID and entity is returned.
+          content:
+            application/json:
+              schema:
+                type: object
+                # add schema for CVE 5.0 format
+        404:
+          description: The requested CVE ID or container has not been populated.
+    put:
+      description: >
+        Updates a given container in a CVE record entry.
+      operationId: updateCVERecordContainerById
+      parameters:
+        - "$ref": "#/components/parameters/api-entity-header"
+        - "$ref": "#/components/parameters/api-secret-header"
+        - name: X-API-AUTHOR
+          in: header
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: The CVE record container data to update
+        content:
+          application/json:
+            schema:
+              # add schema for CVE 5.0 format
+      responses:
+        200:
+          description: the update has been applied
+          content:
+            application/json:
+              schema:
+                # schema for the body
+        400:
+          description: the request body is malformed
+        401:
+          description: the client is not authenticated
+        403:
+          description: >
+            the client is not authorized to update this CVE or a part that was
+            changed. The latter might be caused by a change to a record
+            container for which the authenticated entity is not authorized to
+            change.
+        404:
+          description: >
+            the requested CVE ID has not been previously published.
```

Comment:
This is a potential proposal for an endpoint that could support in-place editing of just the contents of a specific CVE record's container. This may be easier to implement on the server and client sides. I am interested in feedback from folks looking at client and server implementation.